src/backend/core/tests/external_api/metrics/test_external_api_usage_metrics.py

```diff
@@ -0,0 +1,213 @@
+"""
+Tests for the usage metrics external API.
+"""
+
+from django.test import override_settings
+
+import pytest
+from rest_framework.test import APIClient
+from rest_framework_api_key.models import APIKey
+
+from core import factories
+from core.tests.utils.urls import reload_urls
+
+pytestmark = pytest.mark.django_db
+
+# pylint: disable=unused-argument
+
+
+@pytest.fixture(name="api_key_and_key")
+def fixture_api_key_and_key():
+    """
+    Create a API key for the test.
+    The api_key is only available once at creation time, the one used
+    in the requests, where key can be seen as the id of the API key.
+    """
+    api_key, key = APIKey.objects.create_key(name="my-remote-service")
+    return api_key, key
+
+
+@override_settings(METRICS_ENABLED=False)
+def test_usage_metrics_disabled():
+    """
+    Usage metrics should be disabled if the METRICS_ENABLED setting is False.
+    """
+    reload_urls()
+    response = APIClient().get("/external_api/v1.0/metrics/usage/")
+
+    assert response.status_code == 404
+
+
+@override_settings(METRICS_ENABLED=True)
+def test_usage_metrics_no_keys():
+    """
+    Users should not be allowed to retrieve usage metrics without a valid API key.
+    """
+    reload_urls()
+    response = APIClient().get("/external_api/v1.0/metrics/usage/")
+
+    assert response.status_code == 401
+    assert response.json() == {
+        "errors": [
+            {
+                "attr": None,
+                "code": "not_authenticated",
+                "detail": "Authentication credentials were not provided.",
+            },
+        ],
+        "type": "client_error",
+    }
+
+
+@override_settings(METRICS_ENABLED=True)
+def test_usage_metrics_with_valid_key(api_key_and_key):
+    """
+    API keys should be allowed to retrieve usage metrics.
+    """
+    reload_urls()
+    _, key = api_key_and_key
```

Comment:
This does not match with what is explain the docstring of this fixture.
In the tuple returned, the first parameter is the api_key `the one used in the requests` but here you are using the second value returned, the `key`, in the request.